src/IoC/IoC.java

```diff
@@ -0,0 +1,19 @@
+package IoC;
+
+import java.util.Map;
+
+public class IoC {
+    private static IoC instance;
+    private Map<String,Object> myMap;
+
+    public static IoC getInstance() {
+        if (instance == null) {
+            instance = new IoC();
+        }
+        return instance;
+    }
+
+    public <O> O getFromPool(String key) {
```

Comment:
`O` is not conventional for java, rename to `T`

Reply:
Rename method to `get`, it is more generic.

Reply:
You need to check for nulls before casting

Reply:
yes

Reply:
solved